### ISSUE TITLE
Set up syncing graph for losing and gaining access to workspace

### ISSUE THREAD

**Issue by simonrozsival** [toggl/mobileapp#3374]
Depends on #3372 

The order of states during `pull` sync should be this:

- `FetchAllSinceState`
- `DetectLosingAccessToWorkspacesState` (#2899)
- State from #3357 
    - if there are new workspaces -> go to state #3436 and then to  #3358, from this state always loop back to `FetchAllSinceState`
    - else continue to `persistWorkspaces`

**Comment by juanlaube** [comment]
Why do we need to schedule a clean up loop when there are new workspaces? 🤔 

I think it's the opposite, we need to schedule the clean up when losing access to workspaces, and for that we may need a new transition result from `DetectLosingAccessToWorkspacesState` that goes to the clean up state only if needed.

**Comment by simonrozsival** [comment]
You are right.

**Comment by simonrozsival** [comment]
Implementation was merged into a release branch.
